client/src/components/InterviewCard/InterviewCard.jsx

```diff
@@ -64,63 +71,111 @@ const InterviewCard = () => {
     setNote(event.target.value);
   };
 
-  const handleBlur = () => {
-    // TODO Save the note when the input field loses focus
-    saveNote();
-  };
+  const endInterview = () => {
+    let date = new Date();
+    let months = [
+      "01",
+      "02",
+      "03",
+      "04",
+      "05",
+      "06",
+      "07",
+      "08",
+      "09",
+      "10",
+      "11",
+      "12",
+    ];
+    const data = {
+      date: `${date.getFullYear()}-${
+        months[date.getMonth()]
+      }-${date.getDate()}`,
+      student_id: studentId,
+      staff_id: userId,
+      question1_id: 1,
+      rating1: rating1 || 3,
+      question2_id: 2,
+      rating2: rating2 || 3,
+      question3_id: 3,
+      rating3: rating3 || 3,
+      notes: note,
+      rating_score: rating1 + rating2 + rating3,
+    };
 
-  const saveNote = () => {
-    // TODO Implement your save logic here, e.g., make an API call, update state, etc.
-    console.log("Saving note:", note);
-    interviewData.questions[sel];
-  };
-
-  useEffect(() => {
-    getAllQuestionsData();
-  }, []);
-  const getAllQuestionsData = async () => {
-    await fetch(`${routeHTTP}/questions`)
-      .then((response) => response.json())
-      .then((data) => {
-        console.log("data:", data);
-        // setInterviewQuestions(data);
-        // console.log(interviewQuestions)
-      });
+    fetch(`${baseurl}/attempt`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
+    }).then(() =>
+      toast("Interview Attempt saved. Redirecting...", { autoClose: 3000 })
+    );
+    fetch(`${baseurl}/student/${studentId}`, {
```

Comment:
the two back to back fetches aren't ideal so if anyone has some ideas on how to more elegantly do this lmk

Reply:
might be a good use case for Promise.all and using async/await